### PR TITLE
Add loco-cli

### PR DESCRIPTION
**Pull request by SebRut**
## Description of changes

See https://loco.rs/ and https://docs.rs/crate/loco-cli/0.2.6.
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
-  [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/lo/loco-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ lib, rustPlatform, fetchFromGitHub, stdenv }:
+rustPlatform.buildRustPackage rec {
```
Comment (rhendric) [inline pkgs/by-name/lo/loco-cli/package.nix:2]: You don't need `rec` here.

File: pkgs/by-name/lo/loco-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchFromGitHub {
+    owner = "loco-rs";
+    repo = "loco";
+    rev = "51e0362";
+    hash = "sha256-ZiAl+Ru2ggLy7RRqQySwKRbWtGesR7ZgREIpHKrJ00Q=";
+    sparseCheckout = [ "loco-cli" ];
+  };
+
+  cargoLock = { lockFile = ./Cargo.lock; };
```
Comment (rhendric) [inline pkgs/by-name/lo/loco-cli/package.nix:14]:
```suggestion
  cargoLock.lockFile = ./Cargo.lock;
```

File: pkgs/by-name/lo/loco-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  sourceRoot = "source/loco-cli";
+
+  cargoSha256 = lib.fakeSha256;
```
Comment (rhendric) [inline pkgs/by-name/lo/loco-cli/package.nix:22]: Please replace this with the actual hash!

**Review by rhendric — CHANGES_REQUESTED** [review]

File: pkgs/by-name/lo/loco-cli/package.nix (hunk 3 above)
Comment (SebRut) [reply pkgs/by-name/lo/loco-cli/package.nix]: Removed as its not needed with `cargoLock.lockFile`. (see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#importing-a-cargolock-file-importing-a-cargolock-file)

**Review by rhendric — APPROVED** [review]

**Pull request merged**